install/kubernetes/cilium/templates/cilium-configmap.yaml

```diff
@@ -725,10 +725,12 @@ data:
   enable-custom-calls: {{ .Values.customCalls.enabled | quote }}
 {{- end }}
 
-{{- if and .Values.bgp.enabled (not .Values.bgp.announce.loadbalancerIP) }}
+{{- if and .Values.bgp.enabled (and (not .Values.bgp.announce.loadbalancerIP) (not .Values.bgp.announce.podCIDR)) }}
   {{ fail "BGP was enabled, but no announcements were enabled. Please enable one or more announcements." }}
 {{- else if and .Values.bgp.enabled .Values.bgp.announce.loadbalancerIP }}
   bgp-announce-lb-ip: {{ .Values.bgp.announce.loadbalancerIP | quote }}
+{{- else if and .Values.bgp.enabled .Values.bgp.announce.podCIDR }}
```

Comment:
Is it correct that one can only enable bgp-announce-lb-ip or bgp-announce-pod-cidr, but not both ("else if")?

Reply:
Both can be enabled, we should change the template to allow that

Reply:
I think for simplicity I want to do this: 

```
{{- if and .Values.bgp.enabled (and (not .Values.bgp.announce.loadbalancerIP) (not .Values.bgp.announce.podCIDR)) }}
  {{ fail "BGP was enabled, but no announcements were enabled. Please enable one or more announcements." }}
{{- end }}

{{- if and .Values.bgp.enabled .Values.bgp.announce.loadbalancerIP }}
  bgp-announce-lb-ip: {{ .Values.bgp.announce.loadbalancerIP | quote }}
{{- end }}

{{-if and .Values.bgp.enabled .Values.bgp.announce.podCIDR }}
  bgp-announce-pod-cidr: {{ .Values.bgp.announce.podCIDR | quote }}
{{- end}}
```

which takes the form of: 
  guard_feature_configuration
  set_feature_parameter
  set_feature_parameter

where the guard ensures the feature can be enabled without issue.